roofit/roofitcore/inc/LinkDef.h

```diff
@@ -105,7 +105,10 @@
 
 
 // Old LinkDef2.h
-#pragma link C++ namespace RooFit ;
+
+// "namespace RooFit" is in roofit/roofit/inc/Linkdef1.h
+// should not be in the dictionary for two different libraries
+// #pragma link C++ namespace RooFit ;
```

Comment:
Hi, thanks for the fix! But can you add back this line and instead remove it from `roofit/roofit/inc/Linkdef1.h`? Because the `RooFit` namespace is introduced in RooFitCore, and RooFit depends on RooFitCore.

Reply:
The problem is that both RooFit and RooFitCore define symbols in the RooFit namespace.
If "#pragma link C++ namespace RooFit" is in RooFitCore, the autoloading when requested to load the RooFit namespece will then load RooFitCore. However, if the symbol needed was one of the symbols in the RooFit namespace defined in RooFit it will not have been loaded and you get a failure.
If "#pragma link C++ namespace RooFit" is in RooFit, the autoloader when requested to load the RooFit namespace will load RooFit, which will also load RooFitCore, since RooFit depends on RooFitCore. And irrespectively on in which of the libraries the symbol is, it will have been loaded.

Reply:
The best solution would be to use different namespaces in the two libraries, but that would rename things and could cause backward compatibility problems. If the names in one of the libraries are "internal" or "experimental" they can be renamed.

Reply:
Ah ok I get it then, so it has to be like that. Thanks :+1: